packages/grid/x-data-grid/src/components/containers/GridOverlay.tsx

```diff
@@ -35,7 +35,9 @@ const GridOverlayRoot = styled('div', {
   alignSelf: 'center',
   alignItems: 'center',
   justifyContent: 'center',
-  backgroundColor: alpha(theme.palette.background.default, theme.palette.action.disabledOpacity),
+  backgroundColor: theme.vars
+    ? `rgba(${theme.vars.palette.common.backgroundChannel} / ${theme.vars.palette.action.disabledOpacity})`
```

Comment:
This will yield a different result in dark mode. I think we can add another token for `theme.vars.palette.background.defaultChannel` in Material UI.